apps/extension/src/storage/base.ts

```diff
@@ -20,30 +20,53 @@ export interface StorageItem<T> {
 }
 
 type Version = string;
+
+export type MigrationMap<OldState, NewState> = {
+  [K in keyof OldState & keyof NewState]?: (
+    prev: OldState[K],
+    get: <K extends keyof NewState>(key: K) => Promise<NewState[K]>,
```

Comment:
Some migrations (in our case for V3), requires access to querying the store for other fields. Hence, the migration function is now given a `get` querier.

Reply:
resolving as outdated